src/Uno.UI/ts/Windows/UI/Xaml/Application.ts

```diff
@@ -13,19 +14,28 @@
 				}
 			}
 			return null;
-        }
+		}
 
 		public static observeSystemTheme() {
-			if (!this.dispatchThemeChange) {
-				this.dispatchThemeChange = (<any>Module).mono_bind_static_method("[Uno.UI] Windows.UI.Xaml.Application:DispatchSystemThemeChange");
+			if (!Application.dispatchThemeChange) {
+				Application.dispatchThemeChange = (<any>Module).mono_bind_static_method("[Uno.UI] Windows.UI.Xaml.Application:DispatchSystemThemeChange");
 			}
 
-			if (window.matchMedia)
-			{
+			if (window.matchMedia) {
 				window.matchMedia('(prefers-color-scheme: dark)').addEventListener("change", () => {
 					Application.dispatchThemeChange();
 				});
 			}
 		}
-    }
+
+		public static observeVisibility() {
+			if (!Application.dispatchVisibilityChange) {
+				Application.dispatchVisibilityChange = (<any>Module).mono_bind_static_method("[Uno.UI] Windows.UI.Xaml.Application:DispatchVisibilityChange");
+			}
+
+			document.addEventListener("visibilitychange", () => {
+				Application.dispatchVisibilityChange(document.visibilityState == "visible");
```

Comment:
On Safari, it may be required to subscribe to `pagehide` event instead.

Reply:
Just whyyyyy Apple? 😆
![NoAppleNoGIF](https://user-images.githubusercontent.com/1075116/117510153-dbed2980-af8b-11eb-837c-f8e1d8d16e07.gif)